jumanji/environments/packing/flat_pack/reward.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+
+import chex
+import jax
+import jax.numpy as jnp
+
+from jumanji.environments.packing.flat_pack.types import State
+
+
+class RewardFn(abc.ABC):
+    @abc.abstractmethod
+    def __call__(
+        self,
+        state: State,
+        action: chex.Numeric,
+        next_state: State,
+        is_valid: bool,
+        is_done: bool,
+    ) -> chex.Numeric:
+        """Compute the reward based on the current state, the chosen action,
+        whether the action is valid and whether the episode is terminated.
+        """
+
+
+class CellDenseReward(RewardFn):
```

Comment:
This seems like a bad reward to have, just rewarding based on how big your block is? So it'll bias the agent to place bigger blocks first? :thinking: 

I think a reward of `sum(state.grid != 0)` makes more sense

Reply:
I think this is not too bad though? And agents seem to be able to learn with it (it is the default reward). I think your suggestion could be its own reward since it would cumulative over time if I understand correctly.

Reply:
I personnally like this reward definition, summing to 1 if you complete the whole puzzle and favors more-completed puzzles than less-completed puzzles. Indeed, if discounted, this reward favors bigger blocks over smaller ones, but that's more of an agent problem (choose a discount not too low) since the environment is episodic anyway.

Reply:
I agree with this. Shall we then go with keeping this reward as is for now and then keep it as the default?